File: Java/app/src/main/java/dev/ogabek/java/activity/MainActivity.java
package dev.ogabek.java.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import dev.ogabek.java.R;
import dev.ogabek.java.fragment.FirstFragment;
import dev.ogabek.java.fragment.SecondFragment;
import dev.ogabek.java.listener.FirstListener;
import dev.ogabek.java.listener.SecondListener;
import dev.ogabek.java.model.Person;

public class MainActivity extends AppCompatActivity implements FirstListener, SecondListener {

    FirstFragment firstFragment;
    SecondFragment secondFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

    }

    private void initViews() {

        firstFragment = new FirstFragment();
        secondFragment = new SecondFragment();

        getSupportFragmentManager().beginTransaction()
                .replace(R.id.firstFragment, firstFragment)
                .replace(R.id.secondFragment, secondFragment)
                .commit();
    }

    @Override
    public void onFirstSend(Person person) {
        secondFragment.updateData(person);
    }

    @Override
    public void onSecondSend(Person person) {
        firstFragment.updateData(person);
    }
}

File: Kotlin/app/src/main/java/dev/ogabek/kotlin/activity/MainActivity.kt
package dev.ogabek.kotlin.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import dev.ogabek.kotlin.Fragment.FirstFragment
import dev.ogabek.kotlin.Fragment.SecondFragment
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.listener.FirstListener
import dev.ogabek.kotlin.listener.SecondListener
import dev.ogabek.kotlin.model.Person

class MainActivity : AppCompatActivity(), FirstListener, SecondListener {

    private lateinit var firstFragment: FirstFragment
    private lateinit var secondFragment: SecondFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

    }

    private fun initViews() {
        firstFragment = FirstFragment()
        secondFragment = SecondFragment()

        supportFragmentManager.beginTransaction()
            .replace(R.id.firstFragment, firstFragment)
            .replace(R.id.secondFragment, secondFragment)
            .commit()
    }

    override fun onFirstSend(person: Person) {
        secondFragment.updateData(person)
    }

    override fun onSecondSend(person: Person) {
        firstFragment.updateData(person)
    }
}